src/ems_client.ts

```diff
@@ -244,7 +251,14 @@ export class EMSClient {
     this._manifestServiceUrl = config.manifestServiceUrl;
     this._tileApiUrl = config.tileApiUrl;
     this._fileApiUrl = config.fileApiUrl;
-    this._emsVersion = this._getEmsVersion(config.emsVersion);
+
+    if (config.emsVersion?.match(REST_API_REGEX)) {
+      this._emsVersion = config.emsVersion;
```

Comment:
Should we validate `config.emsVersion` using the `_getEmsVersion` method as we do in line 259? If so, we could move setting `this._emsVersion` outside of the conditional.

Reply:
We need to check first the regex because `semver.coerce` is totally fine with dates:

```bash
$ node -e "console.log(require('semver').coerce('2023-10-31').toString())"
2023.0.0
```

In 9d14d1a I moved the logic inside the method and then set the `isRestApi` value by comparing the result with the initial value. Hope this makes things easier to read.